### ISSUE TITLE
Disabling the Facebook App Events SDK for a specific build

### ISSUE THREAD

**Issue by gbyesiltas** [segmentio/analytics-react-native#930]
## Summary

We have set-up the facebook app events plugin as per the docs. For now we want to only enable it in our staging app and not in the actual production app, and not in the app instances we build for End-to-end testing. However, simply not adding the facebook app events plugin doesn't seem to work (at least for Android). 

From what I understand, the FB SDK is automatically initialized for Android. I have tried adding the meta-data that is supposed to turn that off to `AndroidManifest` but that didn't help either.

There is a closed issue about this [here](https://github.com/thebergamo/react-native-fbsdk-next/issues/395) though the specific examples there are related to Expo which doesn't apply to us and honestly the issue was closed without a real conclusion :D

Is there a way to do this?

![Screenshot 2024-03-19 at 17 58 11](https://github.com/segmentio/analytics-react-native/assets/50772478/917fe3a5-0534-4932-a92a-beddda25ff27)
![Screenshot 2024-03-19 at 11 49 07](https://github.com/segmentio/analytics-react-native/assets/50772478/a2a80c63-dc4d-4773-8b1e-5acdea1b55f8)
![Screenshot_1710845354](https://github.com/segmentio/analytics-react-native/assets/50772478/0c7df48d-5d28-41de-b2e1-b372f0842365)

**Comment by oscb** [comment]
Looks like this validation is happening in the FBSDK itself so it's not something we can control from our plugin.
Do you want to fully remove it from the build or just not send any event to FB?

If it's the first one it might be an option in the native FB SDK to control this behavior: https://developers.facebook.com/docs/app-events/overview

If it's fine with just disabling Segment from sending events to FB this one is easier, you can keep the plugin there and add a plugin that filters FB if your app is in debug mode:

```typescript
class FilterFBPlugin extends Plugin {
  type = Plugin.before
  
  execute(event: SegmentEvent) {
    if (!Config.FACEBOOK_APP_ID || !Config.FACEBOOK_CLIENT_TOKEN) {
       event.integrations['Facebook App Events'] = false;
    }
    return event;
  }
}
```

(If FB is automatically sending logs from your app outside of Segment this won't prevent those)

**Comment by gbyesiltas** [comment]
@oscb Hey thanks for the response :)

I have tried adding the meta properties from the FB documentation that I thought could be related to this to disable the auto-initialization, thinking that would fix the issue. But that didn't help unfortunately.

Otherwise, with the second approach was we will need to provide some APP_IDs and CLIENT_TOKENs even for our builds where we don't necessarily want logging and I was concerned about possible side-effects of that (maybe we provide just some mock ones? but again, I assume that may have other side-effects like errors trying to connect to the Facebook SDK).

I feel like at this point I've tried everything that I could think of to disable the initialisation from happening so I may just go with filtering out the events. Thank you for the suggestion!
